src/rules/maxLineLengthRule.ts

```diff
@@ -44,20 +71,71 @@ export class Rule extends Lint.Rules.AbstractRule {
         return `Exceeds maximum line length of ${lineLimit}`;
     }
 
-    public isEnabled(): boolean {
-        return super.isEnabled() && this.ruleArguments[0] as number > 0;
-    }
-
     public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
-        return this.applyWithFunction(sourceFile, walk, this.ruleArguments[0] as number);
+        return this.applyWithFunction(sourceFile, walk, parseOptions(this.ruleArguments));
     }
 }
 
-function walk(ctx: Lint.WalkContext<number>) {
-    const limit = ctx.options;
+function walk(ctx: Lint.WalkContext<Options>) {
     for (const line of getLineRanges(ctx.sourceFile)) {
-        if (line.contentLength > limit) {
-            ctx.addFailureAt(line.pos, line.contentLength, Rule.FAILURE_STRING_FACTORY(limit));
+        if (
+            line.contentLength > ctx.options.limit &&
+            !(
+                ctx.options.ignoreLongUrls &&
+                hasUrl(line, ctx.sourceFile) &&
+                lineLengthMinusUrl(
+                    line,
+                    ctx.sourceFile,
+                    Math.round(ctx.options.limit / 2),
+                ) <= ctx.options.limit
+            )
+        ) {
+            ctx.addFailureAt(
+                line.pos,
+                line.contentLength,
+                Rule.FAILURE_STRING_FACTORY(ctx.options.limit),
+            );
+        }
+    }
+}
+
+function buildOptions(options: Partial<Options>): Options {
+    options.ignoreLongUrls =
+        options.ignoreLongUrls === undefined ? false : options.ignoreLongUrls;
+    options.limit = options.limit === undefined ? DEFAULT_LIMIT : options.limit;
+    return options as Options;
+}
+
+function hasUrl(line: LineRange, sourceFile: ts.SourceFile): boolean {
+    return URL_REGEX.test(sourceFile.text.substr(line.pos, line.contentLength));
+}
+
+function isOptions(arg: number | Options): arg is Options {
+    return (
+        (arg as Options).limit !== undefined ||
+        (arg as Options).ignoreLongUrls !== undefined
+    );
+}
+
+function lineLengthMinusUrl(
+    line: LineRange,
+    sourceFile: ts.SourceFile,
+    minUrlLength: number,
+): number {
+    let longUrlLength = 0;
+    const lineText = sourceFile.text.substr(line.pos, line.contentLength);
+    for (const url of lineText.match(URL_REGEX)!) {
+        if (url.length >= minUrlLength) {
+            longUrlLength += url.length;
         }
     }
+    return lineText.length - longUrlLength;
+}
+
+function parseOptions(args: number[] | Options[]): Options {
```

Comment:
instead of the array, you could pass only the first element. that would make your typeguard work.
the no-unsafe-any failure can be avoided by using `parseOptions(this.ruleArguments[0] as any)`